### ISSUE TITLE
Nodes report incorrect capacity usage

### ISSUE THREAD

**Issue by AhmedHanafy725** [threefoldtech/zos#1508]
env: devnet
node_id: 17

**node statistics**
```ts
[
  {
    ver: 1,
    uid: '19.2',
    cmd: 'zos.statistics.get',
    exp: 0,
    try: 2,
    dat: '{"total":{"cru":24,"sru":512110190592,"hru":9001778946048,"mru":201863462912,"ipv4u":0},"used":{"cru":0,"sru":117300002816,"hru":0,"mru":25017600819,"ipv4u":0}}',
    src: 19,
    dst: [ 56 ],
    ret: 'bb50b1cd-c5d7-4c10-9642-f1961070da3f',
    shm: '',
    now: 1639477667,
    err: ''
  }
]

```

when trying to deploy on this node, got this error `Not enough space left in pools of this type ssd`
contract_id: 2837

```json
{{
   "version":0,
   "twin_id":56,
   "contract_id":2837,
   "metadata":"{\\'testVMs\\': true}",
   "description":"test deploying VMs via ts grid3 client",
   "expiration":1626394539,
   "signature_requirement":{
      "requests":[
         {
            "twin_id":56,
            "required":false,
            "weight":1
         }
      ],
      "weight_required":1,
      "signatures":[
         {
            "twin_id":56,
            "signature":"b099155dcbcf61bde356bb2eedac2ad4fc81d99410b455f7bac7f2fd364f896ec23387d94ad146da0d489b0a2351f1e4b476f199a462f73c7a54c1f46956ea87",
            "signature_type":"sr25519"
         }
      ]
   },
   "workloads":[
      {
         "version":0,
         "name":"wedDisk",
         "type":"zmount",
         "data":{
            "size":96636764160
         },
         "metadata":"{\\'testVMs\\': true}",
         "description":"test deploying VMs via ts grid3 client",
         "result":{
            "created":1639476826,
            "state":"error",
            "message":"failed to find a candidate to host vdisk of size \\'96636764160\\': Not enough space left in pools of this type ssd",
            "data":{
               "volume_id":"56-2837-wedDisk"
            }
         }
      },
      {
         "version":0,
         "name":"testvm",
         "type":"zmachine",
         "data":{
            "flist":"https://hub.grid.tf/tf-official-apps/base:latest.flist",
            "network":{
               "public_ip":"",
               "interfaces":[
                  {
                     "network":"wedtest",
                     "ip":"10.249.2.3"
                  }
               ],
               "planetary":true
            },
            "size":1073741824,
            "compute_capacity":{
               "cpu":1,
               "memory":2147483648
            },
            "mounts":[
               {
                  "name":"wedDisk",
                  "mountpoint":"/testdisk"
               }
            ],
            "entrypoint":"/sbin/zinit init",
            "env":{
               "SSH_KEY":"ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQCa5srzZwh3ulajtY2EZ1SiPY1JNelcaP8O/FZqrnJi6OxAPijl0KzoNrzgemqxhAS/eIglBYbgQuw/Po15MtdMgXmfrtNgrZjQQtLxGFz5KmUbzawPGI7iRkN40LEo0y0hcGLV1G+YiNO+3YU7K5I+gos+04OUJe4HYjcp92nAEviqxa40po2f67KgP5xrZxaOpELZA/hIf1wCzCyTsdvu3k+hw1QlSTIso6WTcUw7LLssvxAs7JZ31kgx+L740xQJWsiVv/go3td0GuETfRSbfjBtOD/wIEHG5UtazOrR+8ukotqQ/ERWuyx1abaEKwro3fLunmjhfgDbnJYy7As1 ahmed@ahmed-Inspiron-3576"
            }
         },
         "metadata":"{\\'testVMs\\': true}",
         "description":"test deploying VMs via ts grid3 client",
         "result":{
            "created":1639476829,
            "state":"error",
            "message":"invalid disk \\'wedDisk\\' state",
            "data":{
               "id":"56-2837-testvm",
               "ip":"10.249.2.3",
               "ygg_ip":"301:a9bd:9b77:ce71:2241:4298:b41d:5013"
            }
         }
      }
   ]
}
```

**Comment by muhamadazmy** [comment]
The `btrfs filesystem df` is reporting a misleading usage info. We need to switch to use `btrfs filesystem useage` instead.

**Comment by muhamadazmy** [comment]
Can you report if this issues happens again, on this or a different node. It's not as easy fix as i thought: While switching to use `btrfs filesystem usage` instead of `filesystem df` is an improvment, we don't actually depend on the usage information to compute free space, instead we depend on `reservations` calculations. This takes into account limits set on volumes, and vdisk files.

**Comment by AhmedHanafy725** [comment]
Tried on other nodes(1, 12, 14), but couldn't reproduce.

**Comment by rawdaGastan** [comment]
Tried on node: 13

Deploying a single vm with disk size = 50 GB failed with error :
`Failed to deploy zmount with name wedDisk on node 13 due to: failed to find a candidate to host vdisk of size '53687091200': not enough space left in pools of this type ssd`

![Screenshot from 2022-09-18 16-11-33](https://user-images.githubusercontent.com/47260239/190911887-2a73b6ec-95fe-4c08-97ab-486dc07a0e00.png)

The logs: 

![Screenshot from 2022-09-18 16-14-40](https://user-images.githubusercontent.com/47260239/190911902-6d89086c-6752-4f2e-ab8b-d24367c2c07a.png)


After checking the resources of the node using `proxy` and `zos`:
The node has free sru after the disk used!

![Screenshot from 2022-09-18 16-24-06](https://user-images.githubusercontent.com/47260239/190911982-6231fde3-c57c-486a-9cfc-ddbb337407a1.png)

**Comment by muhamadazmy** [comment]
I checked the node and I realized there are 2 issues. note that there is one contract that is in grace period on the node.

There are 2 issues here:
- Gridproxy is reporting the wrong "reserved" amount from the contracts. I have reported this before that the grid-porxy did not take the contracts that are in grace period into account. I am not sure if this is fixed or not yet bet seems not fully
- The capacity reported by the node was wrong (for sure) and it's probably because for some reason the capacity used "cached in memory" was out of sync with actual used capacity on the system. Hence I will create an issue to fix this and always recalculate used capacity from the actual system resources rather than track it in memory. (A restart of provisiond fixed it) and the node now reports the correct used capacity.

On the other hand, the grid proxy uses it's knowledge of what active contract on the node to calculate how much resources are avaialble. and it seems the calculations is also wrong. I assume it doesn't take into account (still) contracts in grace period.

Eitherway the machine had the right to reject the deployment because the remaining capacity was not enough

> edit

check the comments below, seems the graceperiod issue on the proxy was indeed fixed, but instead there is a missin 102G from contract 8903. They are reported by the node but don't appear in graphql hence gridproxy doesn't know about them.

**Comment by muhamadazmy** [comment]
I just also found that contract 8093 on devnet while it actually using 204G on the node (13) the grid-porxy is only reporting 102. Looks like from the setup that the deployment has 2 VMs each uses 102 G (2G rootfs, and 100G disk) each.

Still checking if this is a node reporting issue or a chain issue

**Comment by muhamadazmy** [comment]
some debugging shows this:
```2022-09-19T09:25:21Z debug reporting contract usage contract=7447 cru=0 hru=19327352832 mru=0 sru=0
2022-09-19T09:25:21Z debug reporting contract usage contract=7479 cru=0 hru=9663676416 mru=0 sru=0
2022-09-19T09:25:21Z debug reporting contract usage contract=7499 cru=0 hru=9663676416 mru=0 sru=0
2022-09-19T09:25:21Z debug reporting contract usage contract=7573 cru=0 hru=32212254720 mru=0 sru=0
2022-09-19T09:25:21Z debug reporting contract usage contract=8903 cru=4 hru=0 mru=8589934592 sru=219043332096
2022-09-19T09:25:21Z debug reporting contract usage contract=5885 cru=0 hru=0 mru=0 sru=0
2022-09-19T09:25:21Z debug reporting contract usage contract=7102 cru=0 hru=0 mru=0 sru=0
2022-09-19T09:25:21Z debug reporting contract usage contract=5892 cru=0 hru=0 mru=0 sru=0
2022-09-19T09:25:21Z debug reporting contract usage contract=8045 cru=1 hru=0 mru=8589934592 sru=161061273600
2022-09-19T09:25:21Z debug reporting contract usage contract=8912 cru=1 hru=0 mru=2147483648 sru=9114222592
```
by the moment of writing this comment, you can see the node has 2 active contracts. on provisiond start it reports the used capacity to the grid.

For some reason contract 8903 while it reports 219043332096 bytes of SRU (204 G) in graphql it shows ONLY 102 G
![image](https://user-images.githubusercontent.com/10920323/190989381-60c7f4f2-d28e-491c-bca0-f9780d99e017.png)

It's only wrong for this contract, so I was wondering if it's a bad graphql handler for an update contract resources event ?! that doesn't actual reflect the actual data.

This also adds up to the bad results of the capacity planner.

**Comment by muhamadazmy** [comment]
If you add this 102 G missing from the proxy reported sru it will add up to the correct node used capacity. so I guess the grace period issue is actually fixed in the proxy but we have another issue (the missing 102G from contract 8903)

**Comment by DylanVerstraete** [comment]
Fixed in graphql `2.5.14`, waiting on https://github.com/threefoldtech/tf_operations/issues/1164

**Comment by DylanVerstraete** [comment]
Verified fix on devnet
